### DISCUSSION TITLE
The problem of ID sorting under Mongo

### DISCUSSION THREAD

**Discussion by Lydanne** [mikro-orm/mikro-orm#2575] [Q&A]
**Describe the bug**
I want to sort the query by ID, but it has no effect under Mongo.

```ts
em.find({orderBy:{id:-1}});
```

But Debug log
```
[Nest] 59491  - 2021/12/25 下午8:26:24     LOG [MikroORM] [query] db.getCollection('User').find({ deletedAt: { '$eq': null } }, { session: undefined }).sort({ id: -1 }).toArray(); [took 4 ms]
```

This should be_ id

**Versions**

| Dependency | Version |
| - | - |
| node | 14 |
| typescript | ? |
| mikro-orm | 4.5.9 |
| your-driver | mongo |

**Comment by B4nan** [comment]
You need to order by pk, not by serialized pk. 

What is you entity definition?

**Reply by Lydanne** [reply]
```ts
@Entity()
export class User {
  [EntityRepositoryType]?: UserRepo;

  @PrimaryKey()
  _id!: ObjectId;

  @SerializedPrimaryKey()
  id!: string; // string variant of PK, will be handled automatically

  @Property()
  unionid: string;

  @Property({ nullable: false })
  openid: string;

  @IsString()
  @Property()
  avatar: string; 

  @IsString()
  @Property()
  nick: string; 
}
```

This is my entity

**Reply by B4nan** [reply]
Yeah, so as I said, use the pk, which is `_id`

**Reply by Lydanne** [reply]
Why is ID used in filtering and ID used in sorting `_id` ?
Should we be consistent here

**Reply by alexojegu** [reply]
I agree, seems better be consistent here. If it is possible.
